From cf6d431e02b3fa2b1fb0476bc2ead934fd7fe68c Mon Sep 17 00:00:00 2001
From: Chaohong guo <chaohong.guo@intel.com>
Date: Thu, 26 Jul 2018 16:00:24 +0800
Subject: [PATCH] Enable guest OS to boot on VSBL

For booting from VSBL, guest VM should boot from real mode instead of
protected mode.
---
 hypervisor/arch/x86/guest/vcpu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hypervisor/arch/x86/guest/vcpu.c b/hypervisor/arch/x86/guest/vcpu.c
index cd1e645..8912f79 100644
--- a/hypervisor/arch/x86/guest/vcpu.c
+++ b/hypervisor/arch/x86/guest/vcpu.c
@@ -365,7 +365,7 @@ int prepare_vcpu(struct vm *vm, uint16_t pcpu_id)
 		} else {
 #ifdef CONFIG_EFI_STUB
 			/* currently non-vm0 will boot kernel directly */
-			vcpu->arch_vcpu.cpu_mode = CPU_MODE_PROTECTED;
+			vcpu->arch_vcpu.cpu_mode = CPU_MODE_REAL;
 #else
 			vcpu->arch_vcpu.cpu_mode = CPU_MODE_REAL;
 #endif
-- 
2.17.1

